compiler/rustc_infer/src/infer/error_reporting/mod.rs

```diff
@@ -2014,16 +2014,16 @@ impl<'a, 'tcx> InferCtxt<'a, 'tcx> {
                         (ty::Tuple(_), ty::Tuple(_)) => {}
                         (ty::Tuple(_), _) if expected.tuple_fields().count() == 1 => {
                             if let Ok(code) = self.tcx.sess().source_map().span_to_snippet(span) {
-                                if let Some(code) =
-                                    code.strip_prefix('(').and_then(|s| s.strip_suffix(')'))
-                                {
-                                    err.span_suggestion(
-                                        span,
-                                        "use a trailing comma to create a tuple with one element",
-                                        format!("({},)", code),
-                                        Applicability::MaybeIncorrect,
-                                    );
-                                }
+                                let code_stripped = code
+                                    .strip_prefix('(')
+                                    .and_then(|s| s.strip_suffix(')'))
+                                    .unwrap_or(&code);
+                                err.span_suggestion(
+                                    span,
+                                    "use a trailing comma to create a tuple with one element",
+                                    format!("({},)", code_stripped),
+                                    Applicability::MaybeIncorrect,
+                                );
                             }
```

Comment:
While you're at it, could you change this to use a multipart suggestion? Then we don't need to get the source code snippet, and the UI is improved.

It'd look like this (though please make the change in a new commit):

```rust
                            err.multipart_suggestion(
                                "use a trailing comma to create a tuple with one element",
                                vec![
                                    (span.shrink_to_lo(), "(".into()),
                                    (span.shrink_to_hi(), ",)".into()),
                                ],
                                Applicability::MaybeIncorrect,
                            );
```

Reply:
Certainly can, looks much better